analyses/snv-callers/scripts/04-create_consensus_mut_files.R

```diff
@@ -0,0 +1,266 @@
+# If overwrite is set to FALSE, check if these exist before continuing
+if (!opt$overwrite) {
+  # Make a list of the output files
+  output_files <- c(consensus_mut_file, consensus_mut_zip_file, consensus_tmb_file)
+
+  # Find out which of these exist
+  existing_files <- file.exists(output_files)
+
+  # If all files exist; stop
+  if (all(existing_files)) {
+    stop(cat(
+      "Stopping; --overwrite is not being used and all output files already exist
+      in the designated --output directory."
+    ))
+  }
+  # If some files exist, print a warning:
+  if (any(existing_files)) {
+    warning(cat(
+      "Some output files already exist and will not be overwritten unless you use --overwrite: \n",
+      paste0(output_files[which(existing_files)], "\n")
+    ))
+  }
+}
+############################### Read in the files ##############################
+# Read in merged VAF_df
+vaf_df <- readr::read_rds(needed_files[[1]])
+
+# If the caller chosen for the VAF reported isn't in the column, stop.
+if (!(opt$vaf %in% vaf_df$caller)) {
+  stop("Caller chosen with --vaf does not exist in the 'callers column in the
+  master VAF file.")
+}
+# Read in merged TMB file
+tmb_df <- readr::read_rds(needed_files[[2]])
+
+# Read in caller per mutation file
+callers_per_mutation <- readr::read_rds(needed_files[[3]])
+
+# If the file exists or the overwrite option is not being used, do not write the
+# consensus mutation file.
+if (file.exists(consensus_mut_file) && !opt$overwrite) {
+  # Stop if this file exists and overwrite is set to FALSE
+  warning(cat(
+    "A consensus mutation file already exists: \n",
+    consensus_mut_file, "\n",
+    "Use --overwrite if you want to overwrite it."
+  ))
+} else {
+  # Let's get the mutation ids for combination of
+  consen_mutations <- callers_per_mutation %>%
+    dplyr::filter(caller_combo == dplyr::sym(opt$combo)) %>%
+    dplyr::select(-caller_combo) %>%
+    dplyr::pull(mutation_id)
+
+  # Stop if no mutations are found
+  if (length(consen_mutations) == 0) {
+    stop("No mutations had this combination of callers call it. Double check that
+       the combination you specified with --combo is spelled exactly right and
+       the callers are in alphabetical order. ")
+  }
+
+  # Isolate the mutations to only these mutations, use the Strelka2 stats.
+  consen_mutation <- vaf_df %>%
+    dplyr::filter(
+      caller == opt$vaf,
+      mutation_id %in% consen_mutations
+    ) %>%
+    readr::write_tsv(consensus_mut_file)
+}
+
+# If the file exists or the overwrite option is not being used, do not zip the
+# consensus mutation file.
+if (file.exists(consensus_mut_zip_file) && !opt$overwrite) {
+  # Stop if this file exists and overwrite is set to FALSE
+  warning(cat(
+    "A zipped consensus mutation file already exists: \n",
+    consensus_mut_zip_file, "\n",
+    "Use --overwrite if you want to overwrite it."
+  ))
+} else {
+  # Zip this file up.
+  zip(consensus_mut_zip_file, consensus_mut_file)
```

Comment:
I'm not going to ask you to change this but I would have just stuck this in the logic above. If you're generating the new tsv file, you should generate the new zip file.